src/PostgREST/Parsers.hs

```diff
@@ -30,6 +30,12 @@ pRequestSelect :: Text -> Either ApiRequestError [Tree SelectItem]
 pRequestSelect selStr =
   mapError $ parse pFieldForest ("failed to parse select parameter (" <> toS selStr <> ")") (toS selStr)
 
+pRequestOnConflict :: Text -> Either ApiRequestError  [FieldName]
+pRequestOnConflict oncStr =
+  mapError $ parse columns ("failed to parse on_conflict parameter (" <> toS oncStr <> ")") (toS oncStr)
+  where
+    columns = pFieldName `sepBy1` char ','
```

Comment:
You could omit this `where` if you reuse the `pColumns` function:
https://github.com/PostgREST/postgrest/blob/e12c1319b605dea2074ad521b0b74f869da67de4/src/PostgREST/Parsers.hs#L237-L238